portfolio/src/main/webapp/script.js

```diff
@@ -114,34 +119,100 @@ function filterPhotos(filter = null) {
+$('#filter-mountain').click(() => filterPhotos('mountain'));
+$('#filter-none').click(() => filterPhotos());
 
 // eventListeners for sorting photos
-$("#newest").click(() => sortPhotos("newest"));
-$("#oldest").click(() => sortPhotos("oldest"));
+$('#newest').click(() => sortPhotos('newest'));
+$('#oldest').click(() => sortPhotos('oldest'));
+
+/**
+ * Create comment component with location, description, and link
+ * @param {json} comment a json object that contains the data for a comment html component
+ * @return {JQuery}
+ */
+function createComment(comment) {
+  const linkClass = `"rec-link${comment.link ? '' : ' empty-link'}"`;
+  return $(`
+    <div class="comment">
+      <button
+        class="btn btn-block text-left rec-location"
+        id="btn${comment.id}"
+        type="button"
+        data-toggle="collapse"
+        data-target="#rec-${comment.id}"
+        aria-expanded="false"
+        aria-controls="rec-${comment.id}"
+      >
+        <i class="fa fa-caret-right"></i>
+        ${comment.location}
+      </button>
+      <div
+        id="rec-${comment.id}"
+        class="collapse"
+        aria-labelledby="rec-${comment.id}"
+      >
+        <div class="rec-description">
+          <div class="rec-description-txt">
+            ${comment.description || 'No Description.'}
+          </div>
+          <a
+            href=${comment.link}
+            target="_blank"
+            rel="noopener noreferrer"
+            class=${linkClass}
+            >Learn more</a
+          >
+        </div>
+      </div>
+    </div>
+  `);
+}
 
 /**
  * fetch text from /data to display
  */
 function getFetchRequest() {
-  fetch("/data").then(response => response.json()).then((json) => {
-    for (const comment of json) {
-      const component = $(`
-        <div>
-          ${comment.location}
-          <a href=${comment.link} target=\"_blank\">Learn more</a>
-        </div>
-      `)
-      $("#comments").append(component);
-    }
-  });
+  const comments = [];
+  const commentIds = [];
+  fetch('/data')
+    .then((response) => response.json())
+    .then((json) => {
+      for (const comment of json) {
+        const component = createComment(comment);
+        comments.push(component);
+        commentIds.push(`btn${comment.id}`);
+      }
+      $('#comments').empty();
+      $('#comments').append(comments);
+    })
+    .then(() => {
+      for (const cid of commentIds) {
+        const $elem = $(`#${cid}`);
+        $elem.on('click', () => {
+          $elem.find('.fa').toggleClass('rotated');
+          const angle = $elem.find('.fa').hasClass('rotated') ? 90 : 0;
+          $elem.find('.fa').animate(
+            { deg: angle },
+            {
+              duration: 200,
+              step: function (now) {
+                $elem.find('.fa').css({
+                  transform: 'rotate(' + now + 'deg)',
+                });
+              },
+            }
+          );
```

Comment:
Btw, you could do this much easier and performant by using CSS transitions/animations. I wrote a quick pen to demo: https://codepen.io/mecolmg/pen/PoZwVrd